addons/resource_distribution_controller.go

```diff
@@ -156,17 +159,44 @@ func (r *ResourceDistributionReconciler) Reconcile(ctx context.Context, req ctrl
 	}
 	if len(templateList.Items) == 0 {
 		logger.Info("No resources to distribute to StorageConsumers.")
-		err = removeFinalizerFromObject(ctx, r.SpokeClient, &addonDeletionlock, ResourceDistributionFinalizer)
-		if err != nil {
+		addonDeletionlock.Data = nil
+		if err = r.SpokeClient.Update(ctx, &addonDeletionlock); err != nil {
 			return ctrl.Result{}, err
 		}
 		return ctrl.Result{}, nil
 	}
 
-	var storageClientMapping *corev1.ConfigMap
-	storageClientMapping, err = utils.GetStorageClientMapping(ctx, r.SpokeClient, r.CurrentNamespace)
-	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("unable to distribute resources to StorageConsumers: %w", err)
+	if addonDeletionlock.Data == nil {
+		addonDeletionlock.Data = make(map[string]string)
+	}
+
+	var manifestWorkList workv1.ManifestWorkList
+	listOpts := &client.ListOptions{
+		Namespace: r.SpokeClusterName,
+		LabelSelector: labels.SelectorFromSet(map[string]string{
+			utils.CreatedByLabelKey: utils.CreatorMulticlusterOrchestrator,
+		}),
+	}
+	if err = r.HubClient.List(ctx, &manifestWorkList, listOpts); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	tmpAddonDeletionLockData := addonDeletionlock.Data
+	addonDeletionlock.Data = make(map[string]string)
+	for _, mw := range manifestWorkList.Items {
+		if mw.DeletionTimestamp.IsZero() {
+			cId := mw.Labels[utils.CreatedForClientID]
+			if _, ok := addonDeletionlock.Data[cId]; !ok {
+				addonDeletionlock.Data[cId] = ""
+			}
+		}
+	}
```

Comment:
```suggestion
for _, mw := range manifestWorkList.Items {
    if mw.DeletionTimestamp.IsZero() {
        if mw.Labels != nil {
            if cId := mw.Labels[utils.CreatedForClientID]; cId != "" { 
                addonDeletionlock.Data[cId] = ""  
            }
        }
    }
}
```

Reply:
Label nil checks to avoid panic. 
Also we could just directly add cid into the config map if detected.